engine/flink/components/base-tests/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/JavaCollectionsSerializationTest.scala

```diff
@@ -64,6 +73,40 @@ class JavaCollectionsSerializationTest extends AnyFunSuite with FlinkSpec with M
     }
   }
 
+  test("should serialize java map without changing fields order") {
+
+    ResultsCollectingListenerHolder.withListener { collectingListener =>
+      val model = modelData(collectingListener, List(record))
+
+      val sampleMap =
+        ('a' to 'z')
+          .map(x => x -> Random.nextInt())
+          .sortBy(_._2)
+
+      runScenario(
+        model,
+        process(
+          Some(
+            sampleMap
+              .map { case (c, i) => s""""$c" : $i""" }
+              .mkString("{", ",", "}")
+              .spel
+          )
+        )
+      )
+
+      val linkedHashMap = new util.LinkedHashMap[String, AnyRef]()
+      sampleMap.foreach { case (char, int) => linkedHashMap.put(s"$char", Integer.valueOf(int)) }
+
+      val result = collectingListener.results
+        .nodeResults("end")
+        .map(_.variableTyped[Map[_, _]]("mapVariable"))
+
+      result shouldBe List(Some(linkedHashMap))
```

Comment:
Does it check the order? I think that the intent would be more clear if you'd compare lists instead